From abb6fb93f8773bc9feff06e1ea47610fb43b4ff5 Mon Sep 17 00:00:00 2001
From: Michael Pauzner <mpauzner@habana.ai>
Date: Wed, 07 May 2025 14:32:34 +0300
Subject: [PATCH] [SW-228214] gaudi3: serdes: fix wrong polarity

During transition to serdes configuration in flash,
XOR values of tx/rx polarity were lost.

Change-Id: Iaa0945f1d7b5294884143ccbf222e598a783da54
Signed-off-by: Michael Pauzner <mpauzner@habana.ai>
---

diff --git a/src/hl_serdes/gaudi3/serdes.c b/src/hl_serdes/gaudi3/serdes.c
index 6ad8bef..8fcc008 100644
--- a/src/hl_serdes/gaudi3/serdes.c
+++ b/src/hl_serdes/gaudi3/serdes.c
@@ -473,6 +473,10 @@
 	if (rc)
 		goto read_err;
 
+	/* serdes lane polarity inversion mask */
+	module_config.tx_pol ^= HL_SERDES_TX_POLARITY;
+	module_config.rx_pol ^= HL_SERDES_RX_POLARITY;
+
 	/* build link_id config file path and read */
 	sprintf(file_name, "/serdes-%d", link_id);
 	LOG_DBG("SerDes link_id config file: %s", file_name);
